app/models/open_graph_cache.rb

```diff
@@ -33,7 +33,7 @@ def self.find_or_create_by(opts)
   end
 
   def fetch_and_save_opengraph_data!
-    uri = URI.parse(url.start_with?("http") ? url : "http://#{url}")
+    uri = URI.parse(url.start_with?("http") ? url : "https://#{url}")
```

Comment:
This assumes that every URL ever posted supports TLS - which is not a good assumption to make.